pkg/kds/global/components.go

```diff
@@ -250,8 +258,19 @@ func Callbacks(s sync_store.ResourceSyncer, k8sStore bool, kubeFactory resources
 			}
 
 			return s.Sync(rs, sync_store.PrefilterBy(func(r model.Resource) bool {
-				return strings.HasPrefix(r.GetMeta().GetName(), fmt.Sprintf("%s.", clusterName))
+				return r.GetMeta().GetLabels()[mesh_proto.ZoneTag] == clusterName
 			}), sync_store.Zone(clusterName))
 		},
 	}
 }
+
+func isOldZone(rs model.ResourceList) bool {
```

Comment:
The one that doesn't add hashes to resource names (since we're doing it on a "sender" side). Do you think we should rename a func or I can just leave comment

Reply:
Can't figure out a better name. A comment is fine since it's backwards compatible code to be removed.

Please check if we can switch the implem to kds feature flag